src/main/java/Deadline.java

```diff
@@ -0,0 +1,26 @@
+public class Deadline extends Task {
+
+    protected String whenDue;
+
+    public Deadline(String description){
+        super(description);
+        String[] lineComponents = description.split("/",2);
+        String[] descriptionComponents = lineComponents[0].split(" ",2);
+        String[] dueDatePrepositions = lineComponents[1].split(" ",2);
+        this.whenDue = "(" + dueDatePrepositions[0] + ": " + dueDatePrepositions[1] + ")";
+        this.description = descriptionComponents[1] + this.whenDue;
+        this.isDone = false;
+        this.taskLabel = "[D]";
+    }
+    public static int add(String line, Task[] list, int currentNumber) {
```

Comment:
I would recommend adding Javadoc comments before each method in order to convey the purpose of the method.